### PR TITLE
[TOOLS-2217] - Edit t-shirt size in taskboard cards in second level

### PR DESCRIPTION
**Pull request by mauricio-frontend**

### CODE REVIEW

File: core/src/main/resources/static/scripts/taskboard.js [hunk 1]
```diff
@@ -124,6 +125,34 @@ function Taskboard() {
         return findInArray(this.issues, i => i.issueKey === issueKey);
     };
 
+    this.getIssueMetadata = function(issueKey, callback) {
+        $.ajax({
+            url: 'ws/issues/issue-metadata?issueKey=' + issueKey,
+            type: 'get',
+            success: function (data, textStatus, jQxhr) {
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/taskboard.js]: MAJOR Remove the unused function parameter "textStatus". [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
MAJOR Remove the unused function parameter "jQxhr". [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)

File: core/src/main/resources/static/scripts/taskboard.js [hunk 2]
```diff
@@ -124,6 +125,34 @@ function Taskboard() {
...
+            }
+        });
+    };
+
+    this.getFieldOptions = function(selectedIssue, fieldId, callback) {
+        var fieldFromIssue = selectedIssue.fields.filter(function(field) {
+            return (field.id == fieldId);
+        });
+        
+        fieldOptions = fieldFromIssue['0'] ? fieldFromIssue['0'].allowedValues : [];
```
Comment (objective-solutions-user) [inline core/src/main/resources/static/scripts/taskboard.js]: BLOCKER Add the "let", "const" or "var" keyword to this declaration of "fieldOptions" to make it explicit. [rule](http://10.44.1.250:9000/coding_rules#rule_key=javascript%3AS2703)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 3 issues
* ![BLOCKER][BLOCKER] 1 blocker
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail.html [hunk 3]
```diff
@@ -2365,6 +1489,7 @@
                 },
 
                 _cardTapped: function () {
+                    this.fire('iron-signal', {name: 'close-ballparks-pickers'});
```
Comment (taksan) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail.html]: * move this to _closeAllPickers
* remove `_closeTShirtBubble` call and implementation

File: cardboard/src/main/resources/static/elements/taskboard/issue-detail-subtasks.html [hunk 4]
```diff
@@ -0,0 +1,179 @@
+<!--
```
Comment (taksan) [inline cardboard/src/main/resources/static/elements/taskboard/issue-detail-subtasks.html]: I think this file is not used anywhere

File: core/src/main/java/objective/taskboard/data/Issue.java [hunk 5]
```diff
@@ -350,18 +350,30 @@ public String getLastBlockReason() {
...
+            return Collections.emptyList();
+        } else {
+            return list.stream()
+                .map(obj -> verifyTShirtSize(obj.getTshirtCustomFieldId()))
                 .collect(Collectors.toList());
+        }
     }
-    
+
+    private CustomField verifyTShirtSize(String fieldId) {
```
Comment (taksan) [inline core/src/main/java/objective/taskboard/data/Issue.java]: makeTshirtField

File: jira-client/src/main/java/objective/taskboard/jira/client/JiraEditIssue.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                try {
+                    return objectMapper.readValue(objectMapper.writeValueAsString(raw), toClass);
+                } catch (IOException e) {
+                    log.error("Unable to convert", e);
+                    return raw;
+                }
+            };
+        }
+
+        public static final FieldSchemaType string = valueOf("string");
```
Comment (taksan) [inline jira-client/src/main/java/objective/taskboard/jira/client/JiraEditIssue.java]: Do we really need all of theses FieldSchemaType? Most of them doesn't appear to be used anywhere

**Pull request merged**